base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy

```diff
@@ -1029,6 +1030,14 @@ abstract class AbstractAsciidoctorBaseTask extends DefaultTask {
         defaultAttrs
     }
 
+    // must use dynamic invocation to invoke collect method for compatibility with Groovy < 2.5.0
+    @CompileDynamic
+    private Set<String> trimOverridableAttributeNotation(Set<String> attributeKeys) {
+        // remove possible trailing '@' character that is used to encode that the attribute can be overridden
+        // in the document itself
+        attributeKeys.collect { k -> k - ~/@$/ } as Set
```

Comment:
We have a utility to deal with that which will avoid dynamic compilation.
```groovy
Transform.toSet (attributeKeys) { k -> k - ~/@$/ }
```

Reply:
Thanks. good to know. Using `Transform.toSet` now.